search/example-cloud-function/main.py

```diff
@@ -0,0 +1,212 @@
+      ],
+  }
+
+
+def query_vertex_search_api_for_term(search_query):
+  """Performs Vertex AI Search for search_query."""
+  # https://cloud.google.com/generative-ai-app-builder/docs/locations#specify_a_multi-region_for_your_data_store
+  client_options = (
+      ClientOptions(api_endpoint=f"{location}-discoveryengine.googleapis.com")
+      if location != "global"
+      else None
+  )
+  # Create a client
+  client = discoveryengine.SearchServiceClient(client_options=client_options)
+
+  # The full resource name of the search engine serving config
+  # e.g. projects/{project_id}/locations/{location}/dataStores/{data_store_id}/servingConfigs/{serving_config_id}
+  serving_config = client.serving_config_path(
+      project=project_id,
+      location=location,
+      data_store=data_store_id,
+      serving_config="default_config",
+  )
+
+  # Optional: Configuration options for search
+  # Refer to the `ContentSearchSpec` reference for all supported fields:
+  # https://cloud.google.com/python/docs/reference/discoveryengine/latest/google.cloud.discoveryengine_v1.types.SearchRequest.ContentSearchSpec
+  content_search_spec = discoveryengine.SearchRequest.ContentSearchSpec(
+      # For information about snippets, refer to:
+      # https://cloud.google.com/generative-ai-app-builder/docs/snippets
+      snippet_spec=discoveryengine.SearchRequest.ContentSearchSpec.SnippetSpec(
+          return_snippet=True
+      ),
+      extractive_content_spec=discoveryengine.SearchRequest.ContentSearchSpec.ExtractiveContentSpec(
+          max_extractive_answer_count=1
+      ),
+      # For information about search summaries, refer to:
+      # https://cloud.google.com/generative-ai-app-builder/docs/get-search-summaries
+      summary_spec=discoveryengine.SearchRequest.ContentSearchSpec.SummarySpec(
+          summary_result_count=5,
+          include_citations=True,
+          ignore_adversarial_query=True,
+          ignore_non_summary_seeking_query=True,
+      ),
+  )
+
+  # Refer to the `SearchRequest` reference for all supported fields:
+  # https://cloud.google.com/python/docs/reference/discoveryengine/latest/google.cloud.discoveryengine_v1.types.SearchRequest
+  request = discoveryengine.SearchRequest(
+      serving_config=serving_config,
+      query=search_query,
+      page_size=10,
+      content_search_spec=content_search_spec,
+      query_expansion_spec=discoveryengine.SearchRequest.QueryExpansionSpec(
+          condition=discoveryengine.SearchRequest.QueryExpansionSpec.Condition.AUTO,
+      ),
+      spell_correction_spec=discoveryengine.SearchRequest.SpellCorrectionSpec(
+          mode=discoveryengine.SearchRequest.SpellCorrectionSpec.Mode.AUTO
+      ),
+  )
+
+  # Execute the search and return the response as proto.
+  response = client.search(request)
+  # Get the documents as a simple dict for JSON response.
+  documents = [
+      get_document_info(Message.to_dict(result)) for result in response.results
```

Comment:
It might make more sense to send `result` as the original type, then do this conversion in `get_document_info()`.

You should also be able to do the `dict` conversion by doing `discoveryengine.SearchResult.to_dict(result)` or 

`discoveryengine.SearchResponse.SearchResult.to_dict(result)` Then you don't need to import `protobuf.Message`